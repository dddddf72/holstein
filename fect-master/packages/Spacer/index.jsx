import { computed, toRefs } from 'vue'
import { createNameSpace } from '../utils'
const [createComponent] = createNameSpace('Spacer')
import './spacer.less'
export default createComponent({
  props: {
    x: {
      type: [Number, String],
      default: 1,
    },
    y: {
      type: [Number, String],
      default: 1,
    },
    inline: Boolean,
  },
  setup(props, { attrs, slots, emit }) {
    const { x, y, inline } = toRefs(props)

    const getMargin = (num) => {
      if (num <= 0) {
        console.error(`[Spacer Error] ${num} should be greater than 0`)
        return
      }
      if (Number.isNaN(Number(num))) {
        console.error(`[Spacer Error] ${num} should be number`)
        return
      }
      return `calc(${num * 15.25}pt + 1px * ${num - 1})`
    }
    const marginTop = computed(() => getMargin(y.value))
    const marginLeft = computed(() => getMargin(x.value))
    return () => (
      <>
        <span
          {...attrs}
          className={`fect-spacer ${attrs.class ? attrs.class : ''}`}
          style={{
            marginTop: marginTop.value,
            marginLeft: marginLeft.value,
            display: inline.value ? 'inline-block' : 'block',
          }}
        ></span>
      </>
    )
  },
})
