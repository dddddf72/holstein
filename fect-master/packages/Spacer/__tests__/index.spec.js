import { mount } from '@vue/test-utils'

import Spacer from '../index'

describe('Spacer', () => {
  it('shoul be render as element', () => {
    const spacer = mount(Spacer)
    expect(spacer.html()).toMatchSnapshot()
  })

  it('should be support inline style', () => {
    const wrapper = mount(<Spacer inline />)
    // expect(wrapper.find('.fect-spacer').html()).toContain(
    //   '"<span class="fect-spacer " style="display: inline-block;"></span>'
    // )
    expect(wrapper.find('.fect-spacer').attributes('style')).toEqual(
      'display: inline-block;',
    )
  })

  it('should be support x and y', () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <Spacer x={'1'} />
            <Spacer y={'2'} />
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support attrs class', () => {
    const wrapper = mount(<Spacer class="test" />)
    expect(wrapper.find('.fect-spacer').attributes('class')).toEqual(
      'fect-spacer test',
    )
  })
})
