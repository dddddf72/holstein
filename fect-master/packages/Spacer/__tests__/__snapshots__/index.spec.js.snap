// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer shoul be render as element 1`] = `<span class="fect-spacer " style="display: block;"></span>`;

exports[`Spacer should be support x and y 1`] = `<span class="fect-spacer " style="display: block;"></span><span class="fect-spacer " style="display: block;"></span>`;
